python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -271,35 +270,24 @@ def repartition(self, num_partitions: int) -> "SparkXShards":
         if self._get_class_name() == 'pandas.core.frame.DataFrame':
             import pandas as pd
 
-            if num_partitions > self.rdd.getNumPartitions():
-                rdd = self.rdd \
-                    .flatMap(lambda df: df.apply(lambda row: (row[0], row.values.tolist()), axis=1)
-                             .values.tolist()) \
-                    .partitionBy(num_partitions)
-
-                schema = self.get_schema()
-
-                def merge_rows(iter):
-                    data = [value[1] for value in list(iter)]
-                    if data:
-                        df = pd.DataFrame(data=data, columns=schema['columns']) \
-                            .astype(schema['dtypes'])
-                        return [df]
-                    else:
-                        # no data in this partition
-                        return iter
-
-                repartitioned_shard = SparkXShards(rdd.mapPartitions(merge_rows))
-            else:
-                def combine_df(iter):
-                    dfs = list(iter)
-                    if len(dfs) > 0:
-                        return [pd.concat(dfs)]
-                    else:
-                        return iter
+            rdd = self.rdd \
+                .flatMap(lambda df: df.apply(lambda row: (row[0], row.values.tolist()), axis=1)
+                         .values.tolist()) \
+                .partitionBy(num_partitions)
 
-                rdd = self.rdd.coalesce(num_partitions)
-                repartitioned_shard = SparkXShards(rdd.mapPartitions(combine_df))
+            schema = self.get_schema()
+
+            def merge_rows(iter):
+                data = [value[1] for value in list(iter)]
+                if data:
+                    df = pd.DataFrame(data=data, columns=schema['columns']) \
+                        .astype(schema['dtypes'])
+                    return [df]
+                else:
+                    # no data in this partition
+                    return iter
+
+            repartitioned_shard = SparkXShards(rdd.mapPartitions(merge_rows), transient=transient)
```

Comment:
`transient=transient` should be used in all the ` repartitioned_shard` objects in this method